### PR TITLE
Handle gmt set GMT_THEME classic in a modern session

### PR DESCRIPTION
**Pull request by PaulWessel**
Since modern defaults have been set, this PR _meant to be merged into gmt-themes_ will flag all the defaults that differ between classic and modern themes so they are updated.
With this PR, the example

```
gmt begin test_modern2 png
    gmt set GMT_THEME classic
    gmt basemap -R-90/-70/0/20 -JM5c -Bafg -V
gmt end show
```

yields a classic plot.  @meghanrjones, please check if it works as you expected, and what will happen if anyone wants to switch to minimal instead?

### CODE REVIEW

File: src/gmt_init.c [hunk 1]
```diff
@@ -6319,6 +6319,31 @@ GMT_LOCAL void gmtinit_conf_classic (struct GMT_CTRL *GMT) {
...
+		GMT_keyword_updated[GMTCASE_FONT_ANNOT_SECONDARY] = true;
+		GMT_keyword_updated[GMTCASE_FONT_LABEL] = true;
+		GMT_keyword_updated[GMTCASE_FONT_HEADING] = true;
+		GMT_keyword_updated[GMTCASE_FONT_TAG] = true;
+		GMT_keyword_updated[GMTCASE_FONT_TITLE] = true;
+		GMT_keyword_updated[GMTCASE_FONT_SUBTITLE] = true;
+		GMT_keyword_updated[GMTCASE_FONT_LOGO] = true;
+		GMT_keyword_updated[GMTCASE_MAP_ANNOT_OFFSET_PRIMARY] = true;
+		GMT_keyword_updated[GMTCASE_MAP_ANNOT_OFFSET_SECONDARY] = true;
+		GMT_keyword_updated[GMTCASE_MAP_LABEL_OFFSET] = true;
```
Comment (meghanrjones) [inline src/gmt_init.c:6334]:
```suggestion
		GMT_keyword_updated[GMTCASE_MAP_ANNOT_OFFSET_SECONDARY] = true;
		GMT_keyword_updated[GMTCASE_MAP_FRAME_AXES] = true;
		GMT_keyword_updated[GMTCASE_MAP_LABEL_OFFSET] = true;
```

**Comment by meghanrjones** [comment]
Should FORMAT_GEO_MAP be included?

**Comment by PaulWessel** [comment]
> Should FORMAT_GEO_MAP be included?

Does it vary between themes?

File: src/gmt_init.c [hunk 2]
```diff
@@ -6319,6 +6319,31 @@ GMT_LOCAL void gmtinit_conf_classic (struct GMT_CTRL *GMT) {
...
+	if (GMT->current.setting.run_mode == GMT_MODERN) {	/* Means we switch to classic in a modern mode setssion */
+		GMT_keyword_updated[GMTCASE_FONT_ANNOT_PRIMARY] = true;
+		GMT_keyword_updated[GMTCASE_FONT_ANNOT_SECONDARY] = true;
+		GMT_keyword_updated[GMTCASE_FONT_LABEL] = true;
+		GMT_keyword_updated[GMTCASE_FONT_HEADING] = true;
+		GMT_keyword_updated[GMTCASE_FONT_TAG] = true;
+		GMT_keyword_updated[GMTCASE_FONT_TITLE] = true;
+		GMT_keyword_updated[GMTCASE_FONT_SUBTITLE] = true;
+		GMT_keyword_updated[GMTCASE_FONT_LOGO] = true;
+		GMT_keyword_updated[GMTCASE_MAP_ANNOT_OFFSET_PRIMARY] = true;
```
Comment (meghanrjones) [inline src/gmt_init.c:6332]:
```suggestion
		GMT_keyword_updated[GMTCASE_FONT_LOGO] = true;
		GMT_keyword_updated[GMTCASE_FORMAT_GEO_MAP] = true;
		GMT_keyword_updated[GMTCASE_MAP_ANNOT_OFFSET_PRIMARY] = true;
```

**Review by meghanrjones — APPROVED** [review]

File: src/gmt_init.c [hunk 3]
```diff
@@ -3096,8 +3096,8 @@ void gmt_reload_history (struct GMT_CTRL *GMT) {
 }
 
 void gmt_reload_settings (struct GMT_CTRL *GMT) {
-	gmt_conf_SI(GMT);				/* Get the original system defaults */
-	(void)gmt_getdefaults (GMT, NULL);	/* Overload user defaults */
+	gmt_conf_SI(GMT);				/* Get the original system defaults for SI */
+	(void)gmt_getdefaults (GMT, NULL);	/* Overload with any user defaults form initial gmt.conf */
```
Comment (meghanrjones) [inline src/gmt_init.c:3100]:
```suggestion
	(void)gmt_getdefaults (GMT, NULL);	/* Overload with any user defaults from initial gmt.conf */
```

**Pull request merged**